great_expectations/datasource/fluent/data_asset/path/directory_asset.py

```diff
@@ -47,23 +47,29 @@ def add_batch_definition_daily(self, name: str, column: str) -> BatchDefinition:
         # todo: test column
         return self.add_batch_definition(
             name=name,
-            partitioner=ColumnPartitionerDaily(column_name=column),
+            partitioner=ColumnPartitionerDaily(
```

Comment:
These `method_name`s are the default values, but just added so that pydantic serializes correctly, right?

If so, I think our serialization relies on way too much knowledge on how things are initialized, and should change (in another PR)

Reply:
Sounds like we need to do what we do for the `type` field, and update the `self.__fields_set__` attribute before serialization.

https://github.com/great-expectations/great_expectations/blob/dc46c682a66402f6c9b79ad7e67b51d59557366c/great_expectations/datasource/fluent/fluent_base_model.py#L176

https://github.com/great-expectations/great_expectations/blob/dc46c682a66402f6c9b79ad7e67b51d59557366c/great_expectations/datasource/fluent/constants.py#L9-L11